kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -176,21 +185,22 @@ def load(self) -> ir.Table:
             reader = getattr(self.connection, f"read_{self._file_format}")
             return reader(self._filepath, self._table_name, **self._load_args)
         else:
-            return self.connection.table(self._table_name)
+            return self.connection.table(self._table_name, **self._table_args)
 
     def save(self, data: ir.Table) -> None:
         if self._table_name is None:
             raise DatasetError("Must provide `table_name` for materialization.")
 
         writer = getattr(self.connection, f"create_{self._materialized}")
-        writer(self._table_name, data, **self._save_args)
+        writer(self._table_name, data, **self._table_args)
```

Comment:
Is this right? I think the table args should only apply to the `table` call, but haven't looked into it deeply before commenting now.

Reply:
`create_table` & `create_view` both take a database argument as well. If a user wants to write a table to a specific catalog / db schema, they have to provide that argument. Otherwise, it writes to the active catalog / db schema. 

This may be way out of line with what you were thinking, but I was assuming that `save_args` & `load_args` would be dropped from `TableDataset` in version 6 so I didn't want to pass database in either of those. Instead, I combined `table_args` and `save_args` to avoid breaking anything now assuming the API would change a bit in version 6. I'm not sure if there's already a plan for that, but I thought it might look something like this? :man_shrugging: 
```{yaml}
bronze_tracks:
  type: ibis.FileDataset # use `to_<file_format>` (write) & `read_<file_format>` (read)
  connection:
    backend: pyspark
  file_args:
    filepath: hf://datasets/maharshipandya/spotify-tracks-dataset/dataset.csv
    file_format: csv
    materialized: view
    overwrite: True
    table_name: tracks #`to_<file_format>` has no database parameter so there's no ability to write to a specific catalog / db schema atm, `to_<file_format>` just writes to w/e is active
    sep: "," 

silver_tracks:
  type: ibis.TableDataset # would use `create_<materialized>` (write) & `table` (read)
  connection:
    backend: duckdb
    database: data/spotify.db
  table_args:
    name: tracks
    database: spotify.silver
    overwrite: True
```